### DISCUSSION TITLE
How to build dbeaver source bundle with a jre

### DISCUSSION THREAD

**Discussion by fanyify** [dbeaver/dbeaver#13268] [Q&A] [answered]
I am currently using dbeaver 21 and want to build from source and included jre with it. So my build could running on some environment without jre installed.

**Comment by LonwoLonwo** [comment]
Hello @fanyify 
And what questions do you have?
The small wiki page about "build from sources" is here: https://github.com/dbeaver/dbeaver/wiki/Build-from-sources

**Reply by fanyify** [reply]
I wanna build dbeaver from source which has a jre bundled, just like the community version I downloaded from the official site.

**Reply by LonwoLonwo** [reply]
Ok, cool. And what is the question?

**Reply by fanyify** [reply]
My question is How can I do that? Follow that small wiki page, I can build dbeaver from source but no jre bundled, in order to run dbeaver I have to install a openjdk11 first.

**Reply by LonwoLonwo** [reply]
Here https://docs.oracle.com/en/java/javase/11/install/overview-jdk-installation.html#GUID-8677A77F-231A-40F7-98B9-1FD0B48C346A you can read about JDK installation
Do not forget about PATH for JAVA in system properties.
"java -version" command in the command line (if you are using Windows) helps you to understand, that java is installed correctly.
Ant then git clone for DBeaver from GitHub + "mvn package" (https://github.com/dbeaver/dbeaver/wiki/Build-from-sources)

**Comment by fanyify** [comment, marked as answer]
following this instruction: https://www.eclipse.org/justj/?page=documentation. What you need to do is add a feature to your product definition, then modify your pom file. After building finished, it will embedded a openjdk plugin into your product. Finally the dbeaver could be running on some environment that without jre installed.
